cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/SessionManager.java

```diff
@@ -156,7 +156,8 @@ synchronized int numSessions() {
    * @param request the request to close its session.
    */
   synchronized void closeSession(HttpServletRequest request) {
-    HttpSession session = request.getSession();
+    // Response associated with this request has already been flushed; hence, do not attempt to create a new session.
+    HttpSession session = request.getSession(false);
```

Comment:
The reason we are trying to get a session was because we need to remove it from the `_inProgressSessions`.  If the session returns null here we will only remove the session after it expires. An alternative is to store a request to session mapping so we don't need to use getSession(). That said, it would be good to see if we can figure the timing out.

Reply:
Tests in local deployment has shown that the `getSession` will return the session even after the flush; hence, `_inProgressSessions` can remove the returned session.